back/engines/commercial/report_builder/spec/services/report_builder/queries/analytics/active_users_spec.rb

```diff
@@ -6,25 +6,81 @@
   subject(:query) { described_class.new(build(:user)) }
 
   describe '#run_query' do
-    let(:date) { Date.new(2022, 9, 1) }
-    let(:idea) do
+    it 'returns active users' do
+      date = Date.new(2022, 9, 1)
+      project = create(:single_phase_ideation_project)
       create(:dimension_date, date: date)
       create(:dimension_type, name: 'idea', parent: 'post')
+      create(:idea, created_at: date, project: project)
 
-      create(:idea, created_at: date)
-    end
-
-    it 'returns active users' do
-      params = { start_at: date - 1.day, end_at: date + 1.day, project_id: idea.project_id }
+      params = { start_at: date - 1.day, end_at: date + 1.day, project_id: project.id }
       expect(query.run_query(**params)).to eq(
         [
           [{
-            'count_dimension_user_id' => 1,
+            'count_participant_id' => 1,
             'dimension_date_created.month' => '2022-09',
             'first_dimension_date_created_date' => date
           }],
           [{
-            'count_dimension_user_id' => 1
+            'count_participant_id' => 1
+          }]
+        ]
+      )
+    end
+
+    it 'correctly deals with anonymous and deleted users' do
```

Comment:
I don't mind a larger test, but here I think it could be useful to have a bunch of smaller unit tests such as:
- two anonymous participations by the same user = 1 participant
- two anonymous participations by the same deleted user = 1 participant
- two participations by the "same" deleted user = 2 participants

I think this is a nice way to document the counting logic.